blog/content/post/2020-05-14-ipxe_lz_support.md

```diff
@@ -0,0 +1,326 @@
+---
+title: Starting TrenchBoot's Landing Zone from iPXE
+abstract: 'In this article we present support for starting Landing Zone from
+          another bootloader: iPXE. It may not be as featureful as GRUB2, but
+          it has enough juice to start DRTM using images obtained from a remote
+          server'
+cover: /covers/trenchboot-logo.png
+author: krystian.hebel
+layout: post
+published: true
+date: 2020-05-14
+archives: "2020"
+
+tags:
+  - security
+  - coreboot
+  - iPXE
+  - trenchboot
+categories:
+  - Firmware
+  - Security
+
+---
+
+In this article we present support for starting TrenchBoot's Landing Zone from
+another bootloader: iPXE. It may not be as featureful as GRUB2, but it has just
+enough juice to start DRTM using images obtained from a remote server. You will
+lose that fancy graphical menu and initial splash screen, but do you really need
+it?
+
+## Reasoning
+
+We had to develop a quick way of testing new versions of Landing Zone and kernel
+images. Having Linux installed on the platform worked initially, but every now
+and then we managed to break that installation beyond repair and had to start
+from scratch. Also those platforms were used by a group of developers, it wasn't
+always clear in what state the system was. Of course, we could go with full
+[NixOS installation](https://blog.3mdeb.com/2020/2020-03-31-trenchboot-nlnet-lz/#nixos-installation),
+but it is time-consuming and requires physical presence to insert prepared USB
+drive with installer image - due to unusual circumstances (COVID-19 lockdown,
+remote work) such task is no longer trivial.
+
+iPXE can be easily chained - one instance of bootloader (iPXE or a different
+one) can either download or start new iPXE image from a disk. The best part of
+it is that chainloading is supported by unmodified versions of bootloaders - you
+can test it without messing with your current installation. Isn't that a treat?
+
+## Building iPXE binary
+
+Code can be found in our iPXE fork, [landing_zone branch](https://github.com/3mdeb/ipxe/tree/landing_zone).
+It can be build either as a standalone binary or as a coreboot PXE ROM. Let's
+start with the latter one.
+
+#### Embedded in coreboot
+
+Follow [HTTPS enabling tutorial](https://blog.3mdeb.com/2020/2020-05-06-ipxe-https/),
+with two important changes:
+
+* clone from PC Engines repository and switch to proper branch (TODO: tag):
```

Comment:
I guess we don't want the TODO tag? @krystian-hebel

Reply:
Done